Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1213,68 +1205,6 @@ With the following command you can specifically grep for the log output of the a
-(...)
-package:com.awesomeproject
-(...)
-$ adb shell pm path com.awesomeproject
-package:/data/app/com.awesomeproject-1/base.apk
-$ adb pull /data/app/com.awesomeproject-1/base.apk
-```
-
-`apkx` provides an easy method of retrieving an APK's source code via the command line. It also packages `dex2jar` and CFR and automates the extraction, conversion, and decompilation steps. Install it as follows:
-
-```shell
-$ git clone https://github.com/b-mueller/apkx
-$ cd apkx
-$ sudo ./install.sh
-```
-
-This should copy `apkx` to `/usr/local/bin`. Run it on the APK that you want to test as follows:
-
-```shell
-$ apkx UnCrackable-Level1.apk
-Extracting UnCrackable-Level1.apk to UnCrackable-Level1
-Converting: classes.dex -> classes.jar (dex2jar)
-dex2jar UnCrackable-Level1/classes.dex -> UnCrackable-Level1/classes.jar
-Decompiling to UnCrackable-Level1/src (cfr)
-```
-
-If the application is based solely on Java and doesn't have any native libraries (C/C++ code), the reverse engineering process is relatively easy and recovers almost all the source code. Nevertheless, if the code is obfuscated, this process may be very time-consuming and unproductive. This also applies to applications that contain a native library. They can still be reverse engineered, but the process is not automated and requires knowledge of low-level details.
-
-The "Tampering and Reverse Engineering on Android" section contains more details about reverse engineering Android.
-
-##### Automated Static Analysis
-
-You should use tools for efficient static analysis. They allow the tester to focus on the more complicated business logic. A plethora of static code analyzers are available, ranging from open source scanners to full-blown enterprise-ready scanners. The best tool for the job depends on budget, client requirements, and the tester's preferences.
-
-Some static analyzers rely on the availability of the source code; others take the compiled APK as input.
-Keep in mind that static analyzers may not be able to find all problems by themselves even though they can help us focus on potential problems. Review each finding carefully and try to understand what the app is doing to improve your chances of finding vulnerabilities.
-
-Configure the static analyzer properly to reduce the likelihood of false positives. and maybe only select several vulnerability categories in the scan. The results generated by static analyzers can otherwise be overwhelming, and your efforts can be counterproductive if you must manually investigate a large report.
-
-There are several open source tools for automated security analysis of an APK.
-
-- [QARK](https://github.com/linkedin/qark/ "QARK")
-- [Androbugs](https://github.com/AndroBugs/AndroBugs_Framework "Androbugs")
-- [JAADAS](https://github.com/flankerhqd/JAADAS "JAADAS")
-- [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF "MobSF")
-
-For enterprise tools, see the section "Static Source Code Analysis" in the chapter "Testing Tools."
-
```

Comment:
what was reused of this :) ?

Reply:
This whole "Automated Static Analysis" section is now in 5c under the Static Analysis Section